src/service/admin/mod.rs

```diff
@@ -1257,6 +1269,65 @@ impl Service {
                     )
                 }
             },
+            AdminCommand::Acl(AclCommand::Add { mode, hostname }) => {
+                let host = match Host::parse(&hostname) {
+                    Ok(host) => host,
+                    Err(error) => {
+                        return Ok(RoomMessageEventContent::text_plain(format!(
+                            "failed to parse hostname with error {}",
+                            error
+                        )))
+                    }
+                };
+                if let Err(error) = services().acl.add_acl(host.clone(), mode) {
+                    error!(
+                        "encountered {} while trying to add acl with host {} and mode {:?}",
+                        error, host, mode
+                    );
+                    RoomMessageEventContent::text_plain("error, couldn't add acl")
+                } else {
+                    RoomMessageEventContent::text_plain("successfully added ACL")
+                }
+            }
+            AdminCommand::Acl(AclCommand::Remove { hostname }) => {
+                let host = match Host::parse(&hostname) {
+                    Ok(host) => host,
+                    Err(error) => {
+                        return Ok(RoomMessageEventContent::text_plain(format!(
+                            "failed to parse hostname with error {}",
+                            error
+                        )))
+                    }
+                };
+
+                if let Err(error) = services().acl.remove_acl(host.clone()) {
+                    error!(
+                        "encountered {} while trying to remove acl with host {}",
+                        error, host
+                    );
+                    RoomMessageEventContent::text_plain("error, couldn't remove acl")
+                } else {
+                    RoomMessageEventContent::text_plain("successfully removed ACL")
+                }
+            }
+            AdminCommand::Acl(AclCommand::List { filter }) => {
+                let results = services().acl.list_acls(filter);
+                let mut results_html = String::new();
+                results.iter().for_each(|it| {
+                    results_html.push_str(&format!("* {} | {}\n", it.hostname, it.mode.to_emoji()));
+                });
+                RoomMessageEventContent::text_plain(format!(
+                    "
```

Comment:
This formatting when in use looks unusual:

![image](https://github.com/girlbossceo/conduwuit/assets/112147643/8738d6d2-bff3-4a9c-b700-0556e5514df6)

Reply:
fixed it